gaboon/commands/install.py

```diff
@@ -1,49 +1,301 @@
+        # encoding them.
+        # It results in a ConnectionError exception because the actual download url is encoded.
+        # In this case we try to sanitize the version in url and download again.
+        try:
+            _stream_download(download_url, str(repo_install_path.parent), headers)
+        except ConnectionError:
+            download_url = f"https://api.github.com/repos/{org}/{repo}/zipball/refs/tags/{quote(version)}"
+            _stream_download(download_url, str(repo_install_path.parent), headers)
+
+        installed = next(
+            i for i in repo_install_path.parent.iterdir() if i not in existing
+        )
+        shutil.move(installed, repo_install_path)
+
+        # Update versions file
+        if versions_install_path.exists():
+            with open(versions_install_path, "rb") as f:
+                versions_data = tomllib.load(f)
+                versions_data[f"{org}/{repo}"] = version
+                tomli_w.dump(versions_data, f)
+        else:
+            with open(versions_install_path, "w", encoding="utf-8") as f:
+                toml_string = tomli_w.dumps({f"{org}/{repo}": version})
+                f.write(toml_string)
+        _write_dependencies(github_ids, DependencyType.GITHUB)
+
+
+def _get_latest_version(org: str, repo: str, headers: dict) -> str:
+    response = requests.get(
+        f"https://api.github.com/repos/{org}/{repo}/releases/latest", headers=headers
+    )
+    if response.status_code == 200:
+        return response.json()["tag_name"].lstrip("v")
+
+    response = requests.get(
+        f"https://api.github.com/repos/{org}/{repo}/tags?per_page=1", headers=headers
+    )
+    if response.status_code == 200:
+        data = response.json()
+        if data:
+            return data[0]["name"].lstrip("v")
+
+    raise ValueError(f"Unable to determine latest version for {org}/{repo}")
+
+
+def _stream_download(
+    download_url: str, target_path: str, headers: dict[str, str] = REQUEST_HEADERS
+) -> None:
+    response = requests.get(download_url, stream=True, headers=headers)
+
+    if response.status_code == 404:
+        raise ConnectionError(
+            f"404 error when attempting to download from {download_url} - "
+            "are you sure this is a valid mix? https://github.com/brownie-mix"
+        )
+    if response.status_code != 200:
+        raise ConnectionError(
+            f"Received status code {response.status_code} when attempting "
+            f"to download from {download_url}"
+        )
+
+    total_size = int(response.headers.get("content-length", 0))
+    progress_bar = tqdm(total=total_size, unit="iB", unit_scale=True)
+    content = bytes()
+
+    for data in response.iter_content(1024, decode_unicode=True):
+        progress_bar.update(len(data))
+        content += data
+    progress_bar.close()
+
+    with zipfile.ZipFile(BytesIO(content)) as zf:
+        zf.extractall(target_path)
```

Comment:
i think it makes more sense to have a `--dir` flag, otherwise the user is forced to inherit the directory structure of the project.

Reply:
Agreed. This will take a bit of a refactor. Added as an issue. 

https://github.com/Cyfrin/gaboon/issues/22